spark-operator-api/src/main/java/org/apache/spark/k8s/operator/status/AttemptInfo.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.status;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AttemptInfo {
+  @Builder.Default protected final Long id = 0L;
```

Comment:
If we don't want to complex one, let's change this `Long` object to `AtomicLong`.
```
java.util.concurrent.atomic.AtomicLong
```

Reply:
We can use `addAndGet` instead of `id + 1L`.

Reply:
Actually, attempt info and summary are designed to be immutable after creation. Using Atomic & `addAndGet` would change the underlying value and may cause unintentional updates.

In the most recent commit, I remove all setter on `AttemptInfo` and `AttemptSummary`, hope that reduces the confusion